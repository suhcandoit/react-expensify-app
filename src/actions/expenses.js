import database from '../firebase/firebase';

export const addExpense = (expense) => ({ 
    type:'ADD_EXPENSE', 
    expense 
})

export const startAddExpense = (expenseData = {}) => {
    return((dispath) => {
        const {description='', note='', amount=0, createdAt=0} = expenseData;
        const expense = {description, note, amount, createdAt}
        
        return database.ref('expenses').push(expense).then((ref) => {
            dispath(addExpense({
                id: ref.key,
                ...expense}));
        });
    });
};

export const editExpense = (id, updates) => {
    return({
        type:'EDIT_EXPENSE',
        id,
        updates      
    })
}

export const removeExpense = ({id} = {}) => {
    return({
        type:'REMOVE_EXPENSE',
        id
    })
}

// SET_EXPENSES
export const setExpenses = (expenses) => {
    return({
        type:'SET_EXPENSE',
        expenses
    })
}

export const startSetExpenses = () => {
    return((dispath) => {        
        return database.ref('expenses').once('value').then((snapshot) => {
            const expenses = [];
            snapshot.forEach((data) => {
                expenses.push({
                        id: data.key,
                        ...data.val()
                    }
                )
            });
            dispath(setExpenses(expenses));
        })
    });
};
