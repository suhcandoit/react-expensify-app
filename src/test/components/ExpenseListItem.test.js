import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses';

test('Should render ExpenseList items', () => {
    const wrapper = shallow(<ExpenseListItem key={0} {...expenses[0]} />)
    expect(wrapper).toMatchSnapshot()
})