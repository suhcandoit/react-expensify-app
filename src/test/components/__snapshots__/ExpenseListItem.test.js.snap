// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpenseList items 1`] = `
<Link
  className="list-item"
  to="/edit/1"
>
  <div>
    <h3
      className="list-item__title"
    >
      Gum
    </h3>
    <span
      className="list-item__sub-title"
    >
      December 31st, 1969
    </span>
  </div>
  <h3
    className="list-item__data"
  >
    $1.00
  </h3>
</Link>
`;
